### ISSUE TITLE
Test: configure encoding per file type

### ISSUE THREAD

**Issue by bpasero** [Microsoft/vscode#34769]
Refs: https://github.com/Microsoft/vscode/issues/4891

- [x] anyOS - @alexandrudima 
- [x] anyOS - @seanmcbreen 

Complexity: 2

The `files.encoding` setting can now be configured per file type. Verify:
* this works as advertised
* this also works for untitled files when switching the language mode

**Note:** switching the language mode of a saved file (not untitled file) will not change the encoding of the file because that would basically need to trigger a load again of the file and this should be under user control.

**Comment by Sleaker** [comment]
@bpasero - I've tested this a little in my custom plugin:

I have the following in my contributes section of my package.json:

```
        "languages": [{
                "id": "df",
                "extensions": [".bdy", ".utl", ".en", ".fr", ".inc", ".bch", ".frm"],
                "configuration": "./dataflex.language-configuration.json"
            }],
        "configurationDefaults": {
            "[df]": {
                "files.encoding": "cp437"
            }
        }
```

I'm currently using the debug extension for this, and it mostly works. there's one slight hiccup. When the new instance of vscode opens for debugging and it opens a file that was already open in the workspace it defaults to UTF-8 rather than the language-specific default encoding I've specified. To workaround this issue I have to re-open the file (usually just select another file in the project explorer and then reselect the open file) to get it to switch file encodings.

Expected: already open files in the workspace are opened automatically using the extension default encoding when vscode initially launches.
